.github/workflows/tgui_recompile.yml

```diff
@@ -52,3 +61,31 @@ jobs:
         run: |
           git commit -m "TGUI Rebuild" -a || true
           git push
+
+  auto_rebuild:
+    if: ${{ github.event_name == 'push' }}
+    name: Automatic Rebuild
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 25
+          token: ${{ env.TOKEN }}
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '>=12.13'
+
+      - name: Build TGUI
+        run: bin/tgui
+        working-directory: ./tgui
+
+      - name: Commit and Push Build
+        run: |
+          git config --local user.email "${{ env.COMMIT_EMAIL }}"
+          git config --local user.name "${{ env.COMMIT_NAME }}"
+          git pull origin master
+          git commit -m "Automatic TGUI Rebuild [ci skip]" -a || true
+          git push
```

Comment:
While I cannot currently think of a better way to handle this, I genuinely dislike nearly everything about this. It's incredibly dirty and (while somewhat minimal) poses potential security risks. I would far prefer a method that doesn't require a secondary build & commit after a PR is merged

Reply:
I'm afraid that's not possible without performing a full merge of the master branch into the PR, as unless that happens, it's impossible to have an up-to-date build without user intervention

Reply:
Unless you want to trust the action to perform a full merge, that is